vespa/package.py

```diff
@@ -889,6 +890,156 @@ def __init__(
+                ),
+            )
+        )
+
+        #
+        # Add field for doc token ids
+        #
+        if not doc_token_ids_indexing:
+            doc_token_ids_indexing = ["attribute", "summary"]
+        self.schema.add_fields(
+            Field(
+                name=model_config.doc_token_ids_name,
+                type="tensor<float>(d0[{}])".format(
+                    int(model_config.actual_doc_input_size)
+                ),
+                indexing=doc_token_ids_indexing,
+            ),
+        )
+
+        #
+        # Add rank profiles
+        #
+        constants = {"TOKEN_NONE": 0, "TOKEN_CLS": 101, "TOKEN_SEP": 102}
+        if "contants" in kwargs:
+            constants.update(kwargs.pop("contants"))
+
+        functions = [
+            Function(
+                name="question_length",
+                expression="sum(map(query({}), f(a)(a > 0)))".format(
+                    model_config.query_token_ids_name
+                ),
+            ),
+            Function(
+                name="doc_length",
+                expression="sum(map(attribute({}), f(a)(a > 0)))".format(
+                    model_config.doc_token_ids_name
+                ),
+            ),
+            Function(
+                name="input_ids",
+                expression="tokenInputIds({}, query({}), attribute({}))".format(
+                    model_config.input_size,
+                    model_config.query_token_ids_name,
+                    model_config.doc_token_ids_name,
+                ),
+            ),
+            Function(
+                name="attention_mask",
+                expression="tokenAttentionMask({}, query({}), attribute({}))".format(
+                    model_config.input_size,
+                    model_config.query_token_ids_name,
+                    model_config.doc_token_ids_name,
+                ),
+            ),
+            Function(
+                name="token_type_ids",
+                expression="tensor<float>(d0[1],d1[{}])(\n"
+                "    if (d1 < question_length + 2,\n"
+                "        0,\n"
+                "    if (d1 < question_length + doc_length + 3,\n"
+                "        1,\n"
+                "        TOKEN_NONE\n"
+                "    )))".format(model_config.input_size),
+            ),
+            Function(
+                name="logit0",
+                expression="onnx(" + model_id + ").logits{d0:0,d1:0}",
+            ),
+            Function(
+                name="logit1",
+                expression="onnx(" + model_id + ").logits{d0:0,d1:1}",
+            ),
+        ]
+        if "functions" in kwargs:
+            functions.extend(kwargs.pop("functions"))
+
+        summary_features = [
+            "logit0",
+            "logit1",
+            "input_ids",
+            "attention_mask",
+            "token_type_ids",
+        ]
+        if "summary_features" in kwargs:
+            summary_features.extend(kwargs.pop("summary_features"))
+
```

Comment:
Not sure this is something we always want to add. Added logit0 and logit1 functions to summary features involves evaluating the model also during summary fetch time in addition to during ranking. Could this be an option?

Reply:
We now have a `include_model_summary_features` argument in `add_model_ranking` that default to `False`.